[String] add test case for wordwrap method

mgamaltech opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| License       | MIT

Just test case for ```wordwrap``` method

Commit:
[String] add test case for wordwrap method

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,17 @@ public static function provideBytesAt(): array
+    public function testWordwrap()
+    {
+        $instance = static::createFromString('Lorem Ipsum');
+        $actual = $instance->wordWrap(5, '-');
```
wordwrap

mgamal92 replied:
OK i will add dataProvider and more assertion here

mgamal92 replied:
@Nicals  i've added more test cases

nicolas-grekas reviewed (commented):
I think this deserves a few more input/output.
The logic of wordwrap is more than just replacing a space by a dash :)

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,53 @@ public static function provideBytesAt(): array
+    public function testWordwrap($expected, $actual, $length, $break, $cut=false)
+    {
+        $instance = static::createFromString($actual);
+        $actual = $instance->wordWrap($length, $break, $cut);
```
wordwrap (lowercase "w")

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,53 @@ public static function provideBytesAt(): array
+    /**
+     * @dataProvider wordwrapProvider
+     */
+    public function testWordwrap($expected, $actual, $length, $break, $cut=false)
```
cs typo, see fabbot report

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,53 @@ public static function provideBytesAt(): array
+                2,
+                '-',
+            ],
+
```
extra line

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,53 @@ public static function provideBytesAt(): array
+                '-',
+                true
+            ],
+
```
extra line

nicolas-grekas commented on src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php:
```diff
@@ -55,6 +55,53 @@ public static function provideBytesAt(): array
+            ]
+        ];
+    }
+
```
extra line

nicolas-grekas reviewed (commented):
LGTM, just minor issues remaining (see fabbot report)

mgamal92 commented:
i've fixed these issues :)

mpdude committed:
[PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails

Commit:
[Process] - update @throws phpdoc

fancyweb committed:
[HttpKernel][FileLocator] Fix deprecation message

plozmun committed:
[PHPUnitBridge] file_get_contents() expects parameter 3 to be resource

nicolas-grekas committed:
bug #35254 [PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails (mpdude)

This PR was squashed before being merged into the 3.4 branch.

Discussion
----------

[PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       |
| License       | MIT
| Doc PR        |

Some commands executed by the `simple-phpunit` script are not checked for success. For example [here](https://travis-ci.org/twigphp/Twig/jobs/634110681), Composer fails with the message

```
  [InvalidArgumentException]
  Could not find package phpunit/phpunit with version 7.5.* in a version inst
  allable using your PHP version 7.0.25.
```

Yet, the `simple-phpunit` script happily continues, going over failing `chdir()`, `file_get_contents()` and `include()` calls and eventually returns a successful `0` exit code. So CI tests look OK when in fact PHPUnit was not even downloaded.

Commits
-------

576e18561f [PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails

nicolas-grekas committed:
minor #35274 [PHPUnitBridge] file_get_contents() expects parameter 3 to be resource (plozmun)

This PR was squashed before being merged into the 3.4 branch.

Discussion
----------

[PHPUnitBridge] file_get_contents() expects parameter 3 to be resource

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

file_get_contents 3rd parameter (context) expects resource or NULL to ignore them

Commits
-------

a28a42187c [PHPUnitBridge] file_get_contents() expects parameter 3 to be resource

nicolas-grekas committed:
minor #35265 [Process] - update @throws phpdoc (Benhssaein Youssef)

This PR was merged into the 4.3 branch.

Discussion
----------

[Process] - update @throws phpdoc

| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | no
| License       | MIT
| Doc PR        |

Update PHPDoc throws Exceptions on https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Process/Process.php

Commits
-------

2be7029b70 [Process] - update @throws phpdoc

nicolas-grekas committed:
Merge branch '3.4' into 4.3

* 3.4:
  [PHPUnitBridge] file_get_contents() expects parameter 3 to be resource
  [PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails

nicolas-grekas committed:
Merge branch '4.3' into 4.4

* 4.3:
  [Process] - update @throws phpdoc
  [PHPUnitBridge] file_get_contents() expects parameter 3 to be resource
  [PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails

nicolas-grekas committed:
bug #35269 [HttpKernel][FileLocator] Fix deprecation message (fancyweb)

This PR was merged into the 4.4 branch.

Discussion
----------

[HttpKernel][FileLocator] Fix deprecation message

| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Ref https://github.com/symfony/symfony/pull/34886

`$deprecatedPath` is the foreach value so it only works if the last element triggers the deprecation, otherwise the value is wrong.

Commits
-------

18ce8399d2 [HttpKernel][FileLocator] Fix deprecation message

nicolas-grekas committed:
Merge branch '4.4' into 5.0

* 4.4:
  [Process] - update @throws phpdoc
  [PHPUnitBridge] file_get_contents() expects parameter 3 to be resource
  [PHPUnit-Bridge] Fail-fast in simple-phpunit if one of the passthru() commands fails
  [HttpKernel][FileLocator] Fix deprecation message

nicolas-grekas commented:
Thank you @mgamal92.
